pkgs/development/haskell-modules/non-hackage-packages.nix

```diff
@@ -10,6 +10,8 @@ self: super: {
   multi-ghc-travis = throw ("haskellPackages.multi-ghc-travis has been renamed"
     + " to haskell-ci, which is now on hackage");
 
+  dconf2nix = self.callPackage ../tools/haskell/dconf2nix { };
```

Comment:
Rather than trying to build this package as a `non-hackage-package`, it would be easier for us if you were to just release it to Hackage.

If `dconf2nix` was released to Hackage, then the Haskell tooling here in Nixpkgs will pull in the latest version everytime you make a new release to Hackage.

Also, releasing to Hackage generally helps the wider Hackage community, since non-nixpkg users have an easy way to get your package.

Reply:
I know about the trade-offs but if we think about it, this is a tool for Nix users so I'd expect anyone using it to install it via Nix and not by other means.

OTOH I don't want to maintain a Hackage package, that's why I added myself as a maintainer here.

Reply:
If you don't upload to Hackage, it will probably be more work for you to maintain it here in nixpkgs, but as long as you're willing to deal with that extra work, it is fine.

Check out how `haskellPackages.spago` is packaged in nixpkgs, and model the packaging for `dconf2nix` on that.

Reply:
I might consider it in the long run if this doesn't work out but for now I just want to make it available on Nixpkgs as a top level package, so the PR must exist regardless of being on Hackage or not.

I looked at the definition of [vaultenv](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/haskell/vaultenv/default.nix) to have an idea about how to package it but it's also my first time doing it so I'm not really sure what else I'm missing. I'll have a look at Spago too :+1: